### PR TITLE
chore: use analytics from lib

### PR DESCRIPTION
**Pull request by SavvyShah**
- Use `analytics.js` library for analytics.
- Affects `vue` as well as `react`
- I have added two examples for `react` and `vue`

- [Reactivecore PR](https://github.com/appbaseio/reactivecore/pull/148)
- [Notion card](https://www.notion.so/reactivesearch/RSv4-Use-Analytics-js-for-analytics-recording-91f2833fd1e743199acd27877673c844#3d3db0ba36ef4f18842c80c0491c250d) (Loom attached in comments of card)

### CODE REVIEW

File: packages/vue/src/components/ReactiveBase/index.jsx [hunk 1]
```diff
@@ -62,6 +64,7 @@ const ReactiveBase = {
 				this.$props.theme,
 			),
 			store: this.store,
+			$analytics: computed(()=>this.analyticsRef),
```
Comment (bietkul) [inline packages/vue/src/components/ReactiveBase/index.jsx]: @SavvyShah What are you doing here with `computed`?

File: packages/vue/src/components/ReactiveBase/index.jsx [hunk 2]
```diff
@@ -201,6 +204,12 @@ const ReactiveBase = {
 			if (this.$props.transformResponse) {
 				appbaseRef.transformResponse = this.$props.transformResponse;
 			}
+			const parsedUrl = this.url && this.url.replace(/\/\/.*@/, '//');
```
Comment (bietkul) [inline packages/vue/src/components/ReactiveBase/index.jsx]: @mohdashraf010897 Can you handle `endpoint` here?

File: packages/vue/src/components/ReactiveBase/index.jsx [hunk 3]
```diff
@@ -201,6 +204,12 @@ const ReactiveBase = {
...
+			this.analyticsRef = AppbaseAnalytics.init({
+				index: appbaseRef.app,
+				credentials: appbaseRef.credentials,
+				url: parsedUrl,
```
Comment (bietkul) [inline packages/vue/src/components/ReactiveBase/index.jsx]: @mohdashraf010897 Can you handle `endpoint` here for analytics, only use the host part of `endpoint` if present?

File: packages/web/src/components/basic/ReactiveBase.js [hunk 4]
```diff
@@ -206,6 +208,11 @@ class ReactiveBase extends Component {
 			if (this.props.transformRequest) return this.props.transformRequest(modifiedRequest);
 			return modifiedRequest;
 		};
+		const analyticsRef = AppbaseAnalytics.init({
```
Comment (bietkul) [inline packages/web/src/components/basic/ReactiveBase.js]: @mohdashraf010897 Handle endpoint URL

**Review by bietkul — CHANGES_REQUESTED** [review]
## Functionality Issues
- In your react demo, impressions are not present in request payload.
alt img
- Do you set `customEvents` in analytics instance? Custom events can be configured in ReactiveBase component using `appbaseConfig` prop. If yes, then share a demo. If not, then implement it and then share demo.
- Can you share a demo for Vue which covers all the test scenarios? The demo that you share only covers the click event for ReactiveMap.

File: packages/vue/src/components/ReactiveBase/index.jsx (hunk 1 above)
Comment (SavvyShah) [reply packages/vue/src/components/ReactiveBase/index.jsx]: The properties in provide are not reactive by default. So I followed [this guide.](https://vuejs.org/guide/components/provide-inject.html#working-with-reactivity)

File: packages/vue/src/components/ReactiveBase/index.jsx [hunk 5]
```diff
@@ -201,6 +202,32 @@ const ReactiveBase = {
...
+			};
+
+			try {
+				if (this.$props.endpoint && this.$props.endpoint.url) {
+					// Remove parts between '//' and first '/' in the url
+					analyticsInitConfig.url = this.$props.endpoint.url.replace(
+						/\/\/(.*?)\/.*/,
+						'//$1',
+					);
+					const headerCredentials = this.$props.endpoint.headers.Authorization;
```
Comment (bietkul) [inline packages/vue/src/components/ReactiveBase/index.jsx]: @SavvyShah Add `null` check

File: packages/vue/src/components/ReactiveBase/index.jsx [hunk 6]
```diff
@@ -201,6 +202,32 @@ const ReactiveBase = {
...
+					analyticsInitConfig.credentials = headerCredentials.replace('Basic ', '');
```
Comment (bietkul) [inline packages/vue/src/components/ReactiveBase/index.jsx]: Add `null` check

File: packages/vue/src/components/ReactiveBase/index.jsx [hunk 7]
```diff
@@ -201,6 +202,32 @@ const ReactiveBase = {
 			if (this.$props.transformResponse) {
 				appbaseRef.transformResponse = this.$props.transformResponse;
 			}
+			const analyticsInitConfig = {
+				url: url && url.replace(/\/\/.*@/, '//'),
+				credentials: appbaseRef.credentials,
+				// When endpoint prop is used index is not defined, so we use _default
+				index: appbaseRef.app || '_default',
+				globalCustomEvents: this.$props.appbaseConfig.customEvents,
```
Comment (bietkul) [inline packages/vue/src/components/ReactiveBase/index.jsx]: Add `null` check before reading values

**Review by bietkul — APPROVED** [review]

**Pull request merged**